server/service/src/invoice/outbound_shipment/update/mod.rs

```diff
@@ -231,6 +233,49 @@ mod test {
         let context = service_provider.context().unwrap();
         let service = service_provider.invoice_service;
 
+        //CannotReverseInvoiceStatus
+        assert_eq!(
+            service.update_outbound_shipment(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipment| {
+                    r.id = mock_outbound_shipment_picked().id;
+                    r.status = Some(UpdateOutboundShipmentStatus::Allocated);
+                })
+            ),
+            Err(ServiceError::CannotReverseInvoiceStatus)
+        );
+        //InvoiceDoesNotExist
+        assert_eq!(
+            service.update_outbound_shipment(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipment| { r.id = "invalid".to_string() })
+            ),
+            Err(ServiceError::InvoiceDoesNotExist)
+        );
+        //InvoiceIsNotEditable
+        assert_eq!(
+            service.update_outbound_shipment(
+                &context,
+                &mock_store_a().id,
+                inline_init(|r: &mut UpdateOutboundShipment| {
+                    r.id = mock_outbound_shipment_b().id
```

Comment:
You can add an assert to ensure mock_outbound_shipment_b() has the correct status, e.g. so that if somebody edits it the test case is not broken.

Reply:
hmm, this assert will need to check for quite a few statuses (shipped, delivered, verified), I think we either rename the mock data to also have 'shipped' or create invoice here in test ?

Reply:
[eec66b7](https://github.com/openmsupply/open-msupply/pull/329/commits/eec66b73f3253a4e6a885c9e01bd509478fe5203)